helix-core/src/test/java/org/apache/helix/integration/multizk/TestMultiZkConectionConfig.java

```diff
@@ -225,18 +220,6 @@ public void afterClass() throws Exception {
         System.clearProperty(MetadataStoreRoutingConstants.MSDS_SERVER_ENDPOINT_KEY);
       }
     }
-
-    boolean status = false;
-    try {
-      status = ThreadLeakageChecker.afterClassCheck(testClassName);
-    } catch (Exception e) {
-      LOG.error("ThreadLeakageChecker exception:", e);
-    }
-    // todo: We should fail test here once we achieved 0 leakage and remove the following System print
-    if (!status) {
-      System.out.println(
-          "---------- Test Class " + testClassName + " thread leakage detected! ---------------");
-    }
```

Comment:
About thread leakage: back in 2020 there was a huge effort to crack down thread leak, and I believe those code is a part of that effort. Now that we're removing it, how do we ensure that there's no thread leakage? What's the strategy here?

Reply:
Thanks for sharing the history @NealSun96!

Reply:
Thanks for the comments!
With all due respect, I feel either this tool doesn't work well, or we have serious thread leak problems. Right now one CI ([example](https://pipelines.actions.githubusercontent.com/serviceHosts/7ea409d0-72c7-46ee-906c-148950b1480d/_apis/pipelines/1/runs/10241/signedlogcontent/2?urlExpires=2022-10-25T18%3A34%3A12.7529931Z&urlSigningMethod=HMACV1&urlSignature=RsB0PvBcFCUHv%2F5G7nRtcsRCGOTjEYuk%2FL1Rlvi5Uzw%3D)), this has 295 instances of `thread leakage detected!`. And there is no action taken.
Also I don't see much value of printing this to System.out without failing the test, right now it's just informative, and to be honest, I simply ignore them.
My take is, we can probably keep the class, but remove it from runtime execution until we have fully resolved the current issues (either the code or the tooling).
Hope that makes sense.